templates/react-native/src/services/template.ts.twig

```diff
@@ -0,0 +1,185 @@
+     {
+        super(client);
+     }
+{% for method in service.methods %}
+
+    /**
+     * {{ method.title }}
+     *
+{% if method.description %}
+{{ method.description|comment2 }}
+{% endif %}
+     *
+{% for parameter in method.parameters.all %}
+     * @param {{ '{' }}{{ parameter | getPropertyType(method) | raw }}{{ '}' }} {{ parameter.name | caseCamel | escapeKeyword }}
+{% endfor %}
+     * @throws {{ '{' }}{{ spec.title | caseUcfirst}}Exception}
+     * @returns {% if method.type == 'webAuth' %}{void|string}{% elseif method.type == 'location' %}{URL}{% else %}{Promise}{% endif %}
+
+    */
+    {% if method.type != 'location' and method.type != 'webAuth'%}async {% endif %}{{ method.name | caseCamel }}{{ method.responseModel | getGenerics(spec) | raw }}({% for parameter in method.parameters.all %}{{ parameter.name | caseCamel | escapeKeyword }}{% if not parameter.required or parameter.nullable %}?{% endif %}: {{ parameter | getPropertyType(method) | raw }}{% if not loop.last %}, {% endif %}{% endfor %}{% if 'multipart/form-data' in method.consumes %}, onProgress = (progress: UploadProgress) => {}{% endif %}): {{ method | getReturn(spec) | raw }} {
+{% for parameter in method.parameters.all %}
+{% if parameter.required %}
+        if (typeof {{ parameter.name | caseCamel | escapeKeyword }} === 'undefined') {
+            throw new {{spec.title | caseUcfirst}}Exception('Missing required parameter: "{{ parameter.name | caseCamel | escapeKeyword }}"');
+        }
+
+{% endif %}
+{% endfor %}
+        const apiPath = '{{ method.path }}'{% for parameter in method.parameters.path %}.replace('{{ '{' }}{{ parameter.name | caseCamel | escapeKeyword }}{{ '}' }}', {{ parameter.name | caseCamel | escapeKeyword }}){% endfor %};
+        const payload: Payload = {};
+
+{% for parameter in method.parameters.query %}
+        if (typeof {{ parameter.name | caseCamel | escapeKeyword }} !== 'undefined') {
+            payload['{{ parameter.name }}'] = {{ parameter.name | caseCamel | escapeKeyword }};
+        }
+
+{% endfor %}
+{% for parameter in method.parameters.body %}
+        if (typeof {{ parameter.name | caseCamel | escapeKeyword }} !== 'undefined') {
+            payload['{{ parameter.name }}'] = {{ parameter.name | caseCamel | escapeKeyword }};
+        }
+
+{% endfor %}
+        const uri = new URL(this.client.config.endpoint + apiPath);
+{% if method.type == 'location' or method.type == 'webAuth' %}
+{% if method.auth|length > 0 %}
+{% for node in method.auth %}
+{% for key,header in node|keys %}
+        payload['{{header|caseLower}}'] = this.client.config.{{header|caseLower}};
+
+{% endfor %}
+{% endfor %}
+{% endif %}
+
+        for (const [key, value] of Object.entries(Service.flatten(payload))) {
+            uri.searchParams.append(key, value);
+        }
+{% endif %}
+{% if method.type == 'webAuth' %}
+        if (typeof window !== 'undefined' && window?.location) {
+            window.location.href = uri.toString();
+        } else {
```

Comment:
This will [throw an exception in React Native](https://github.com/appwrite/sdk-for-react-native/issues/10). Ideally, we make it work, but for now, maybe we can just the url.

Reply:
throwing exception

Reply:
Why throw? Why can't we return the url?

Reply:
What will not solve the issue, it will not work as of now so throwing exception.

Reply:
Ok we can tackle this later.